### PR TITLE
Feature/chat message notification

### PR DESCRIPTION
**Pull request by DavidSurkov**
image
image
image
image

### CODE REVIEW

File: src/pages/Chat/Chat.tsx [hunk 1]
```diff
@@ -1,42 +1,19 @@
...
+import ChatRoomsListItem from "pages/Chat/components/ChatRoomsList/ChatRoomsListItem";
 import { NavLink, Route, Routes } from "react-router-dom";
 import ChatRoom from "pages/Chat/components/ChatRoom/ChatRoom";
 import { CHAT_ROOM_ROUTE } from "utils/consts/routeConsts";
-
-export interface IRoomUsers {
-  freelancer: IChatUser;
-  client: IChatUser;
-}
+import useChatData from "./hooks/useChatData";
```
Comment (xx-TOOL-xx) [inline src/pages/Chat/Chat.tsx]: absolute path

**Review by xx-TOOL-xx — APPROVED** [review]

File: src/pages/Chat/Chat.tsx (hunk 1 above)
Comment (DavidSurkov) [reply src/pages/Chat/Chat.tsx]: fixed

File: src/components/navbar/Navbar.tsx [hunk 2]
```diff
@@ -50,10 +42,17 @@ function Navbar() {
...
 
-  const socket = getSocket();
-  socket.on("first-message", () => {
-    setCountNotifications(countNotifications + 1);
-  });
+  useEffect(() => {
+    setTotal(() => {
+      if (
+        (countNotifications && countNotifications.message) ||
+        (countNotifications && countNotifications.other)
```
Comment (lyudad) [inline src/components/navbar/Navbar.tsx]: `countNotifications?.message || countNotifications?.other ` will in not work?

File: src/pages/Chat/hooks/useChatData.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  const { data: notificationsResponse } = useGetMessagesNotificationsQuery();
+
+  const getRoomNotifications = (roomId: number): number[] | undefined => {
+    if (notificationsResponse && notificationsResponse.notifications) {
+      return notificationsResponse.notifications
+        .filter((notification) => notification.roomId === roomId)
+        .map((notification) => notification.id);
+    }
+    return undefined;
```
Comment (lyudad) [inline src/pages/Chat/hooks/useChatData.ts:24]: can we just `return` without any value?

File: src/pages/Notification/Notifications.tsx [hunk 4]
```diff
@@ -1,65 +1,56 @@
...
-          onClick={() => handleReadNotification(item.id || 0)}
-        >
-          <Typography>Notification type: {item.type}</Typography>
-          <Typography>From: {item.fromUser?.firstName}</Typography>
-        </S.NotificationBox>
-      ))}
-    </>
+    <Container>
+      <h1>{t("Notifications.title")}</h1>
+      {data?.notifications?.length ? (
```
Comment (lyudad) [inline src/pages/Notification/Notifications.tsx]: nonono better 
```
!data?.notifications return  <div>{t("Notifications.noNotifications")}</div>
  return (
    <Container>
      <h1>{t("Notifications.title")}</h1>
....
```
if make more simple example instead
```
if(isEqual){
 calculations
calculations
calculations
calculations
calculations
calculations
calculation1
} else return 1
```

better make
```
if (!isEqual) return 1
 calculations
calculations
calculations
calculations
calculations
calculations
calculation1
```

its match more readable and understandable

File: src/services/notifications/chatTypes.ts [hunk 5]
```diff
@@ -40,15 +41,28 @@ export interface ISendMessage {
...
+  message: string;
+}
+export interface INotification {
+  id: number;
+  isRead: boolean;
+  type: NotificationTypeEnum;
+  text: string;
+  createdAt: Date;
+  roomId: number;
+  message: INotificationContent | null;
```
Comment (lyudad) [inline src/services/notifications/chatTypes.ts:55]: `message?: INotificationContent` better make like that

**Comment by lyudad** [comment]
Снимок экрана 2022-10-13 в 21 00 07

**Comment by lyudad** [comment]
Снимок экрана 2022-10-13 в 21 00 46

**Comment by lyudad** [comment]
Снимок экрана 2022-10-13 в 21 01 41
* such
*

File: src/pages/Chat/hooks/useChatData.ts (hunk 3 above)
Comment (DavidSurkov) [reply src/pages/Chat/hooks/useChatData.ts:24]: I believe it is more explicit. as long as the badge takes either number or undefined.

**Comment by lyudad** [comment]
Снимок экрана 2022-10-13 в 21 02 53

File: src/services/notifications/chatTypes.ts (hunk 5 above)
Comment (DavidSurkov) [reply src/services/notifications/chatTypes.ts:55]: message?: INotificationContent === message: INotificationContent | undefined
in my case I can receive INotificationContent or null. more type safe.

**Comment by DavidSurkov** [comment]
> 

send Icon. Or You think its not obvious for use?

**Comment by DavidSurkov** [comment]
> Снимок экрана 2022-10-13 в 21 02 53

If you mean that the text of a message is not shown the is that because I hide it if the notification is being read

File: src/components/navbar/Navbar.tsx (hunk 2 above)
Comment (DavidSurkov) [reply src/components/navbar/Navbar.tsx]: fixed

File: src/pages/Notification/Notifications.tsx (hunk 4 above)
Comment (DavidSurkov) [reply src/pages/Notification/Notifications.tsx]: done

**Comment by DavidSurkov** [comment]
image

File: src/pages/Chat/components/ChatRoomsList/ChatRoomsListItem.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+import { Badge } from "antd";
+import { useEffect } from "react";
+import { useReadNotificationsMutation } from "services/notifications/setNotificationsAPI";
+
+interface IChatUsersList {
+  jobName: string;
+  lastMessage: string;
+  opponentsNameAndAvatar: { name: string; avatar: string };
+  isSelected: boolean;
+  roomNotifications: number[] | undefined;
```
Comment (lyudad) [inline src/pages/Chat/components/ChatRoomsList/ChatRoomsListItem.tsx]: can we use `roomNotifications?: number[]` instead?

**Pull request merged**